packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -484,7 +563,7 @@ class _DraggableScrollableSheetScrollController extends ScrollController {
       physics: physics,
       context: context,
       oldPosition: oldPosition,
-      extent: extent,
+      controller: this,
```

Comment:
This is all private so it's probably not a big deal, but it seems wrong to give a position knowledge of the controller.

It would probably be better to have an abstract class like

```dart
abstract class _DraggableSheetExtentHolder {
  _DraggableSheetExtent extent;
}
```

and make the controller implement that, to make sure that no one comes along later and starts tinkering with controller API in the position and messes something up.

Reply:
What about providing an anonymous function that returns extent? This feels a little lighter weight and I think makes the intent more obvious (to provide mutable state). However, the extent holder seems fine too.

I just pushed the function version, let me know if you prefer the holder version and I'll switch to it.